src/irmin-pack/unix/ext.ml

```diff
@@ -316,6 +318,19 @@ module Maker (Config : Conf.S) = struct
             | Some { use_auto_finalisation = true; _ } ->
                 let* _ = finalise_exn ~wait:false t in
                 Lwt.return_unit
+
+          let latest_live_commit t =
+            let open Result_syntax in
+            let control = File_manager.control t.fm in
+            let pl = Control.payload control in
+            let off = pl.commit_offset_in_prefix in
+            if Int63.Syntax.(off = Int63.zero) then Ok None
+            else
+              let entry =
+                Commit.CA.read_and_decode_entry_prefix ~off t.dispatcher
+              in
+              let+ key = commit_key_from_hash t entry.hash in
```

Comment:
`entry` contains enough informations to create a `Direct` key

Reply:
indeed, thanks